packages/hardhat-viem/src/internal/type-extensions.ts

```diff
@@ -30,16 +31,24 @@ declare module "hardhat/types/runtime" {
       getTestClient(
         testClientConfig?: Partial<TestClientConfig>
       ): Promise<TestClient>;
-      deployContract(
-        contractName: string,
-        constructorArgs?: any[],
-        config?: DeployContractConfig
-      ): Promise<GetContractReturnType>;
-      getContractAt(
-        contractName: string,
-        address: Address,
-        config?: GetContractAtConfig
-      ): Promise<GetContractReturnType>;
+      deployContract: typeof deployContract;
+      sendDeploymentTransaction: typeof sendDeploymentTransaction;
+      getContractAt: typeof getContractAt;
     };
   }
 }
+
+declare module "hardhat/types/artifacts" {
+  // eslint-disable-next-line @typescript-eslint/no-empty-interface
+  interface ArtifactsMap {}
+
+  interface Artifacts {
+    readArtifact<ArgT extends keyof ArtifactsMap>(
+      contractNameOrFullyQualifiedName: ArgT
+    ): ArtifactsMap[ArgT];
+
+    readArtifactSync<ArgT extends keyof ArtifactsMap>(
```

Comment:
Should this one return a Promise?

Reply:
I guess you refer to `readArtifact` and not `readArtifactSync` and it makes a valid point. I see that the generic type definition for `readArtifact` indeed returns a promise: `public async readArtifact(name: string): Promise<Artifact>`. I'm going to change it and test that everything still works.

Reply:
You were right, good catch! 🚀